lib/setopt.c

```diff
@@ -1366,14 +1375,14 @@ static CURLcode setopt_long(struct Curl_easy *data, CURLoption option,
     data->set.upkeep_interval_ms = arg;
     break;
   case CURLOPT_MAXAGE_CONN:
-    if(arg < 0)
+    if(!setopt_long_as_pos_timediff(arg))
       return CURLE_BAD_FUNCTION_ARGUMENT;
-    data->set.maxage_conn = arg;
+    data->set.conn_max_idle_ms = (timediff_t)arg;
     break;
   case CURLOPT_MAXLIFETIME_CONN:
-    if(arg < 0)
+    if(!setopt_long_as_pos_timediff(arg))
       return CURLE_BAD_FUNCTION_ARGUMENT;
-    data->set.maxlifetime_conn = arg;
+    data->set.conn_max_age_ms = (timediff_t)arg;
```

Comment:
Did you forget to multiply with 1000 here? These options are documented to take number of *seconds* in their arguments.